src/element.c

```diff
@@ -56,18 +56,18 @@ struct stumpless_element *
 stumpless_add_param( struct stumpless_element *element,
                      struct stumpless_param *param ) {
   struct stumpless_param **new_params;
-  size_t old_params_size;
-  size_t new_params_size;
+  size_t old_params_count;
```

Comment:
it appears that you can get rid of this local variable now, as the size calculation code that used it no longer needs it.

Reply:
Ok: I've removed these variables and the "+ 1" modification that went to new_params_size (let me know if this change is what is expected).

I've just refactored the 3 commits of this PR by doing git rebase.

Reply:
The plus one is needed to reallocate the array with space for the new param, which is probably why some of the tests are failing. You can probably just use `element->param_count +1` as the second argument to `realloc_array` to fix this.

Reply:
Hello! 

Right: I've just changed the second argument of this `realloc_array` call to `param_count + 1`.